src/lib/feed.ts

```diff
@@ -37,103 +13,45 @@ export type ParsedCursor = ReturnType<typeof parseCursor>;
 
 export interface RankLinksOptions {
   limit: number;
-  cursor?: ParsedCursor;
+  cursor?: number;
   range?: string;
 }
 
 export async function rankLinks({
   limit,
-  cursor = { startTime: undefined, index: undefined },
+  cursor,
   range = "1 day",
 }: RankLinksOptions) {
-  const defaultStartTime = getStartTime();
-  const startTime = cursor.startTime ?? defaultStartTime;
-  const cacheKey = `${startTime}/${range}`;
-  const cached = db
-    .prepare(`SELECT value FROM cache WHERE key = ?`)
-    .get(cacheKey) as { value: string } | undefined;
-
-  let items: [string, PostWithData[]][];
-
-  if (cached) {
-    console.log("CACHE HIT", cacheKey);
-    items = JSON.parse(cached.value);
-  } else {
-    console.log("CACHE MISS", cacheKey);
-    const minTime = `DATETIME('${startTime}', '-${range}')`;
-    const maxTime = `DATETIME('${startTime}')`;
-
-    const posts = db
-      .prepare(
-        `
-          SELECT 
-              post.did,
-              post.rkey,
-              post.url,
-              post.createdAt,
-              COUNT(CASE WHEN reaction.type = 'like' THEN 1 END) AS likes,
-              COUNT(CASE WHEN reaction.type = 'repost' THEN 1 END) AS reposts,
-              COUNT(CASE WHEN reaction.type = 'comment' THEN 1 END) AS comments
-          FROM 
-              post
-              LEFT JOIN reaction ON post.rkey = reaction.rkey AND post.did = reaction.did
-          WHERE
-              post.createdAt >= ${minTime}
-              AND post.createdAt <= ${maxTime}
-              AND (reaction.createdAt >= ${minTime} AND post.createdAt <= ${maxTime} OR reaction.createdAt IS NULL)
-          GROUP BY
-              post.did, post.rkey, post.url, post.createdAt;
-        `
-      )
-      .all() as PostWithData[];
-
-    const top: Record<string, PostWithData[]> = {};
-
-    for (let i = 0; i < posts.length; i++) {
-      const post = posts[i];
-
-      if (!post) {
-        continue;
-      }
-
-      if (!top[post.url]) {
-        top[post.url] = [];
-      }
-
-      top[post.url]?.push(post);
-    }
-
-    items = Object.entries(top)
-      .sort((a, b) => sumPostScore(b[1]) - sumPostScore(a[1]))
-      .map(([url, posts]): [string, PostWithData[]] => [
+  const posts = cacheDb
+    .prepare(
+      `
+      SELECT
+        did,
+        rkey,
         url,
-        posts.sort((a, b) => getPostScore(b) - getPostScore(a)),
-      ])
-      .slice(0, 1500);
-
-    db.prepare(`INSERT OR REPLACE INTO cache (key, value) VALUES (?, ?)`).run(
-      cacheKey,
-      JSON.stringify(items)
-    );
-  }
-
-  if (cursor.index) {
-    items = items.slice(cursor.index + 1);
-  }
-
-  items = items.slice(0, limit);
+        createdAt,
+        score
+      FROM
+        post
+      WHERE
+        createdAt >= DATETIME('now', '-${range}')
```

Comment:
Does this preserve the functionality? In the past implementation the cursor would go into the same page results because of the timestamp. I'm not seeing how we discern between a person scrolling through 4:00pm's feed vs 4;30pm's feed

Reply:
Well, the cache is populated every 10 minutes (you can change that to whatever makes sense to you) but it will stay consistent for that timeframe as it doesn't change until the cron job runs again.

You could conceivably change the logic that creates a new "view" every time the cron runs that stores when it was run. Then on the query side you could find the closest cached view based on what time it is now, and then set that as part of the cursor, then that user could have a locked view for however long you'd want to persist that cached view. I haven't tested this specifically but it seems like it should work.

Reply:
Yeah that's exactly what I was thinking. It shouldn't make it too more complicated. it just changes it to 

```
createdAt >= DATETIME('${cursorTime}', '-${range}')
```

Reply:
So the way I solved this, to keep with the consistent view was to add the write time when writing the cached feed view. This allows you to get the closest written time when reading without a cursor, and then we incorporate that write time as part of the cursor for subsequent feed requests. This will ensure they have a consistent feed view for up to 24 hours before that feed is removed, though if it was removed they'd get the next oldest one that is closest to whatever cursor time they have, but at that point the `createdAt` filter would likely trigger the end of the feed (by not sending anything, well I'm not entirely sure what happens at the app side and how it handles ending of the feed).

This could be an odd experience for the hourly feed. They'd have a feed that would be locked to their view, but it would become stale if they tried to view it beyond an hour. Not entirely sure if we should keep that view consistent until they refresh or if we should let the cutoff like the above 24 hour cutoff to just end the feed. It may be better for UX to just keep it consistent once they are in the feed.

I'm testing a light alteration based on the above to see if it works as expected.

Reply:
Okay I added some additional handling for the hourly feed (it should lock you into the feed you're viewing for up to 24 hours before the cache purges that view). I also fixed logic on the cursor so it maxes the index to the last record and doesn't change after once you hit the last page and there are no more records.

I couldn't find anything official about how it handles the end of a feed sadly, kind of hand wavey around that.

Reply:
I think the feed just ending at a set amount of posts is fine. some links shouldn't see the feed 😅